pkg/exchange/bybit/types.go

```diff
@@ -21,19 +31,108 @@ type WebSocketEvent struct {
 	Args   []string `json:"args"`
 }
 
-func (w *WebSocketEvent) IsValid() error {
+func (w *WebSocketOpEvent) IsValid() error {
 	switch w.Op {
 	case WsOpTypePing:
 		// public event
 		if (w.Success != nil && !*w.Success) ||
 			(w.RetMsg != nil && WsOpType(*w.RetMsg) != WsOpTypePong) {
-			return fmt.Errorf("unexpeted response of pong: %#v", w)
+			return fmt.Errorf("unexpeted response of pong: %+v", w)
 		}
 		return nil
 	case WsOpTypePong:
 		// private event
 		return nil
+	case WsOpTypeSubscribe:
+		if w.Success != nil && !*w.Success {
+			return fmt.Errorf("unexpected subscribe result: %+v", w)
+		}
+		return nil
 	default:
-		return fmt.Errorf("unexpected op type: %#v", w)
+		return fmt.Errorf("unexpected op type: %+v", w)
+	}
+}
+
+type TopicType string
+
+const (
+	TopicTypeOrderBook TopicType = "orderbook"
+)
+
+type DataType string
+
+const (
+	DataTypeSnapshot DataType = "snapshot"
+	DataTypeDelta    DataType = "delta"
+)
+
+type WebSocketTopicEvent struct {
+	Topic TopicType `json:"topic"`
+	Type  DataType  `json:"type"`
+	// The timestamp (ms) that the system generates the data
+	Ts   types.MillisecondTimestamp `json:"ts"`
+	Data interface{}                `json:"data"`
+}
+
+// PriceVolumeSlice represents a slice of price and value.
+//
+// index 0 is Bid/Ask price.
+// index 1 is Bid/Ask size. The *delta data* has size=0, which means that all quotations for this price have been filled or cancelled
+type PriceVolumeSlice [2]fixedpoint.Value
```

Comment:
use `PriceVolume` to avoid confusion, because we already have some types for this kind of data:

![image](https://github.com/c9s/bbgo/assets/50894/785c8be4-0bad-4e89-9608-2c8377794d22)